zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/handlers/IncidentHandler.java

```diff
@@ -127,6 +135,51 @@ public String getIndexName() {
     return indexName;
   }
 
+  private String buildTreePath(final Record<IncidentRecordValue> record) {
+
+    final IncidentRecordValue value = record.getValue();
+    final List<List<Long>> elementInstancePath = value.getElementInstancePath();
+    final List<Integer> callingElementPath = value.getCallingElementPath();
+    final List<Long> processDefinitionPath = value.getProcessDefinitionPath();
+
+    final Long processInstanceKey = Long.valueOf(value.getProcessInstanceKey());
+
+    // example of how the tree path is built when current instance is on the third level of calling
+    // hierarchy:
+    // PI_<parentProcessInstanceKey>/FN_<parentCallActivityId>/FNI_<parentCallActivityInstanceKey>/
+    // PI_<secondLevelProcessInstanceKey>/FN_<secondLevelCallActivityId>/FNI_<secondLevelCallActivityInstanceKey>/
+    // PI_<currentProcessInstanceKey>/FN_<flowNodeId>/FNI_<flowNodeInstanceId>
+    final TreePath treePath = new TreePath();
+    for (int i = 0; i < elementInstancePath.size(); i++) {
+      final List<Long> keysWithinOnePI = elementInstancePath.get(i);
+      treePath.appendProcessInstance(keysWithinOnePI.get(0));
+      if (keysWithinOnePI.get(0).equals(processInstanceKey)) {
+        // when we reached current processInstanceKey, we build the last peace of tree path
+        treePath
+            .appendFlowNode(value.getElementId())
+            .appendFlowNodeInstance(value.getElementInstanceKey());
+        break;
+      } else {
+        final var callActivityId =
+            ProcessCacheUtil.getCallActivityId(
+                processCache, processDefinitionPath.get(i), callingElementPath.get(i));
+        if (callActivityId.isPresent()) {
+          treePath.appendFlowNode(callActivityId.get());
+        } else {
+          LOGGER.debug(
+              "No process found in cache. TreePath won't contain proper callActivityId. processInstanceKey: {}, processDefinitionKey: {}, incidentKey: {}",
+              processInstanceKey,
+              processDefinitionPath.get(i),
+              record.getKey());
+          treePath.appendFlowNode(String.valueOf(callingElementPath.get(i)));
```

Comment:
❓ how are we making any use of this `callingElementIndex` later on if we didn't find it in the cache now?

Reply:
As agreed, I increase the logging level to warning and will be checking how Operate code will deal with such cases, when instead of `callActivityId` we have it's callActivityIndex.